### PR TITLE
Hybrid Token Pool Fixes

### PR DESCRIPTION
**Pull request by jhweintraub**

### CODE REVIEW

File: contracts/gas-snapshots/automation.gas-snapshot [hunk 1]
```diff
@@ -1,13 +1,102 @@
-AutomationForwarder_forward:testBasicSuccess() (gas: 87630)
```
Comment (RensR) [inline contracts/gas-snapshots/automation.gas-snapshot]: Please undo these changes. You should only run `make snapshot` with the ccip profile enabled to avoid things like this

File: contracts/src/v0.8/ccip/pools/USDC/HybridLockReleaseUSDCTokenPool.sol [hunk 2]
```diff
@@ -34,8 +34,8 @@ contract HybridLockReleaseUSDCTokenPool is USDCTokenPool, USDCBridgeMigrator {
   error LanePausedForCCTPMigration(uint64 remoteChainSelector);
   error TokenLockingNotAllowedAfterMigration(uint64 remoteChainSelector);
 
-  /// bytes4(keccak256("NO_CTTP_USE_LOCK_RELEASE"))
-  bytes4 public constant LOCK_RELEASE_FLAG = 0xd43c7897;
```
Comment (RensR) [inline contracts/src/v0.8/ccip/pools/USDC/HybridLockReleaseUSDCTokenPool.sol:38]: Can we share this flag between pools instead of copy-pasting to ensure they use the same one

File: contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    _applyChainUpdates(address(s_pool));
+  }
+}
+
+contract BurnMintWithLockReleaseFlagTokenPool_lockOrBurn is BurnMintWithLockReleaseFlagTokenPoolSetup {
+  function test_Setup_Success() public view {
+    assertEq(address(s_burnMintERC677), address(s_pool.getToken()));
+    assertEq(address(s_mockRMN), s_pool.getRmnProxy());
+    assertEq(false, s_pool.getAllowListEnabled());
+    assertEq("BurnMintTokenPool 1.5.0", s_pool.typeAndVersion());
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol]: Should this not be different?

File: contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  function test_PoolBurn_CorrectReturnData_Success() public {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol]: please follow the test naming convention by replacing `PoolBurn` with `lockOrBurn`

File: contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol (hunk 3 above)
Comment (jhweintraub) [reply contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol]: Yea it's a little strange now that I look at it. I've removed it.

File: contracts/gas-snapshots/automation.gas-snapshot (hunk 1 above)
Comment (jhweintraub) [reply contracts/gas-snapshots/automation.gas-snapshot]: For reasons unexplainable by me the automation tests CI stopped working out of nowhere

File: contracts/src/v0.8/ccip/pools/USDC/HybridLockReleaseUSDCTokenPool.sol (hunk 2 above)
Comment (jhweintraub) [reply contracts/src/v0.8/ccip/pools/USDC/HybridLockReleaseUSDCTokenPool.sol:38]: Sure but it makes the code a little ugly since we have to declare the constant outside of the contract but in the file to make it available.

File: contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol (hunk 4 above)
Comment (RyanRHall) [reply contracts/src/v0.8/ccip/test/pools/BurnMintWithLockReleaseFlagTokenPool.t.sol]: I think this comment still applies, yeah?

**Review by RyanRHall — APPROVED** [review]
LGTM, just the one pending comment about test names
